devito/types/dimension.py

```diff
@@ -709,10 +711,42 @@ class ConditionalDimension(DerivedDimension):
     is_Conditional = True
 
     def __init_finalize__(self, name, parent, factor=None, condition=None,
-                          indirect=False):
+                          indirect=False, **kwargs):
         super().__init_finalize__(name, parent)
+
+        # condition = indexify_expr(condition)
+        # import pdb; pdb.set_trace()
```

Comment:
NOTE FOR REVIEWERS:
The duplicate code here needs of course to be removed and we should be calling `indexify_expr`.
The fact that there is no `subdomains` for types like And, CondEq needs to be fixed. Will be added as second part of this PR. The code below is left to show that it works this way.

Reply:
> The fact that there is no subdomains for types like And, CondEq needs to be fixed

People won't understand what you're talking about

Reply:
Partially explained in comment below.